hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -86,74 +96,74 @@ public void configure() {
                 .log("Validation exception response: ${body}")
                 .handled(true)
                 .id("ValidationException");
-        
-        setupSSLConextPharmanetRegistry(getContext());
-        String pharmNetUrl = String.format(pharmanetUri + "?bridgeEndpoint=true&sslContextParameters=#%s&authMethod=Basic&authUsername=%s&authPassword=%s", SSL_CONTEXT_PHARMANET, pharmanetUser, pharmanetPassword);
-        String basicToken = buildBasicToken(pharmanetUser, pharmanetPassword);
-        log.info("Using pharmNetUrl: " + pharmNetUrl);
-        
-        String isFileDropsEnabled = properties.getValue(IS_FILEDDROPS_ENABLED);
 
         from("jetty:http://{{hostname}}:{{port}}/{{endpoint}}").routeId("hnsecure-route")
 
-			//this route is only invoked when the original route is complete as a kind
+			// This route is only invoked when the original route is complete as a kind
 			// of completion callback.The onCompletion method is called once per route execution.
-			//Making it global will generate two response file drops.
+			// Making it global will generate two response file drops.
 			.onCompletion().modeBeforeConsumer().onWhen(header(Exchange.HTTP_RESPONSE_CODE).startsWith(HTTP_STATUS_OK_CODES)).id("Completion")
-			//creating filedrops if enabled
+				// create filedrops if enabled
 		    	.choice().when(header("isFileDropsEnabled").isEqualToIgnoreCase(Boolean.TRUE))
 		    		.bean(ResponseFileDropGenerater.class).id("ResponseFileDropGenerater")
 				.end()
-		    		//encoding response before sending to consumer
-		    		.setBody().method(new Base64Encoder()).id("Base64Encoder")
-		    		.setBody().method(new ProcessV2ToJson()).id("ProcessV2ToJson")
+				//encoding response before sending to consumer
+				.setBody().method(new Base64Encoder()).id("Base64Encoder")
+				.setBody().method(new ProcessV2ToJson()).id("ProcessV2ToJson")
 			.end()
-			
-			// here the original route continues
+
         	.log("HNSecure received a request")
+			// If a transaction ID is provided in the HTTP request header, use it as the exchange id instead of the camel generated id
+			.choice()
+				.when(header(HTTP_REQUEST_ID_HEADER))
+					.process(exchange -> {
+						exchange.setExchangeId(exchange.getIn().getHeader(HTTP_REQUEST_ID_HEADER, String.class));
+					}).id("SetExchangeIdFromHeader")
+			.end()
```

Comment:
This is the only real change to the route, the rest is just some formatting and spelling cleanup